python/cudf/cudf/core/index.py

```diff
@@ -2926,6 +2925,10 @@ def as_index(arbitrary, **kwargs) -> Index:
         return TimedeltaIndex(arbitrary, **kwargs)
     elif isinstance(arbitrary, CategoricalColumn):
         return CategoricalIndex(arbitrary, **kwargs)
+    elif isinstance(arbitrary, IntervalColumn):
+        return IntervalIndex(arbitrary, **kwargs)
+    elif isinstance(arbitrary, StructColumn):
+        return IntervalIndex(arbitrary, **kwargs)
```

Comment:
Is this really what we want to do? What happens to arbitrary struct columns?
```
s = cudf.Series([{'a': 3, 'b': 'x'}, {'a': 4, 'b': 'y'}])
cudf.core.index.as_index(s._column) # ?
```

Reply:
I don't think we need to do this anymore. I wasn't actually sure what to return for an index version of a struct column. I'm not sure we have one. Anyway, I'll drop this line for the moment.